packages/synapse-react-client/src/components/widgets/query-filter/CombinedRangeFacetFilter.test.tsx

```diff
@@ -0,0 +1,339 @@
+import React from 'react'
+import {
+  ColumnModel,
+  ColumnTypeEnum,
+  FacetColumnResultRange,
+} from '@sage-bionetworks/synapse-types'
+import {
+  CombinedRangeFacetFilter,
+  CombinedRangeFacetFilterProps,
+} from './CombinedRangeFacetFilter'
+import { VALUE_NOT_SET } from '../../../utils/SynapseConstants'
+import { RangeSliderProps } from '../RangeSlider/RangeSlider'
+import { QueryVisualizationContextProvider } from '../../QueryVisualizationWrapper'
+import { Range, RangeProps } from '../Range'
+
+let capturedOnChange:
+  | ((range: { min: string | number; max: string | number }) => void)
+  | undefined
+
+jest.mock('../Range', () => ({
+  Range: jest.fn((props: RangeProps) => {
+    capturedOnChange = props.onChange
+    return <div data-testid="Range"></div>
+  }),
+}))
+jest.mock('../RangeSlider/RangeSlider', () => ({
+  RangeSlider: jest.fn((props: RangeSliderProps) => {
+    capturedOnChange = props.onChange
+    return <div data-testid="RangeSlider"></div>
+  }),
+}))
+
+jest.mock('@mui/material', () => ({
+  Collapse: jest.fn(props => (
+    <div data-testid="Collapse">{props.children}</div>
+  )),
+}))
+
+const mockCallback = jest.fn(() => null)
+const intFacetResult: FacetColumnResultRange = {
+  columnMax: '1999',
+  columnMin: '1996',
+  columnName: 'Year',
+  concreteType: 'org.sagebionetworks.repo.model.table.FacetColumnResultRange',
+  facetType: 'range',
+}
+
+const minColumnNotSetFacetResult: FacetColumnResultRange = {
+  columnMax: '50',
+  columnMin: '10',
+  columnName: 'Value',
+  selectedMax: VALUE_NOT_SET,
+  selectedMin: VALUE_NOT_SET,
+  concreteType: 'org.sagebionetworks.repo.model.table.FacetColumnResultRange',
+  facetType: 'range',
+}
+
+const minRangeFacetResult = {
+  ...minColumnNotSetFacetResult,
+  selectedMin: '10',
+  selectedMax: '40',
+}
+
+const maxColumnNotSetFacetResult: FacetColumnResultRange = {
+  columnMax: '70',
+  columnMin: '30',
+  columnName: 'Value',
+  selectedMax: VALUE_NOT_SET,
+  selectedMin: VALUE_NOT_SET,
+  concreteType: 'org.sagebionetworks.repo.model.table.FacetColumnResultRange',
+  facetType: 'range',
+}
+
+const maxRangeFacetResult = {
+  ...maxColumnNotSetFacetResult,
+  selectedMin: '35',
+  selectedMax: '60',
+}
+
+const columnModel: ColumnModel = {
+  columnType: 'INTEGER',
+  facetType: 'range',
+  id: '86423',
+  name: 'Year',
+}
+
+function createTestProps(
+  overrides?: CombinedRangeFacetFilterProps,
+): CombinedRangeFacetFilterProps {
+  return {
+    facetResults: [intFacetResult, intFacetResult],
+    columnType: columnModel.columnType,
+    label: columnModel.name,
+    onChange: mockCallback,
+    ...overrides,
+  }
+}
+
+describe('CombinedRangeFacetFilter tests', () => {
+  let props: CombinedRangeFacetFilterProps
+  function init(overrides?: CombinedRangeFacetFilterProps) {
+    props = createTestProps(overrides)
+    return render(
+      <QueryVisualizationContextProvider
+        queryVisualizationContext={{
```

Comment:
Type check enforced here led me to adding mocks for all of the required properties (which the RangeFacetFilter test does not need because it's still in the /test/ folder, I think).  Is there another way to do this, since this seems excessive.

Reply:
I want to add a mock helper for this context